File: frontend/gen-pdf.py
"""Generate sample-receipt.pdf with actual S-mark logo and dark theme."""
from reportlab.lib.pagesizes import A4
from reportlab.pdfgen import canvas
from reportlab.lib.colors import HexColor
from PIL import Image
import io

W, H = A4
pub = '/root/bunche/frontend/public'

# Load logo
logo_img = Image.open(f'{pub}/pdf-icon.png').convert('RGBA')
# Save as temp PNG for ReportLab
logo_path = '/tmp/styx-icon.png'
logo_img.save(logo_path)

# Build PDF
c = canvas.Canvas(f'{pub}/sample-receipt.pdf', pagesize=A4)

# ── Background: solid dark ──
c.setFillColor(HexColor('#0a0a0a'))
c.rect(0, 0, W, H, fill=1, stroke=0)

# ── Top accent bar ──
c.setFillColor(HexColor('#0AD25A'))
c.rect(0, H - 8, W, 8, fill=1, stroke=0)

# ── Logo (top-left) ──
c.drawImage(logo_path, 40, H - 80, width=48, height=48, mask='auto')

# ── Brand name (right of logo) ──
c.setFillColor(HexColor('#FFFFFF'))
c.setFont('Helvetica-Bold', 22)
c.drawString(100, H - 55, 'styxproxy')

c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica', 9)
c.drawString(100, H - 70, 'Anonymous Proxy Service')

# ── Right side: receipt label ──
c.setFillColor(HexColor('#0AD25A'))
c.setFont('Helvetica-Bold', 10)
c.drawRightString(W - 40, H - 50, 'PAYMENT RECEIPT')

c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica', 9)
c.drawRightString(W - 40, H - 65, 'styxproxy.com')
c.drawRightString(W - 40, H - 78, f'Issued: {__import__("datetime").date.today().strftime("%B %d, %Y")}')

# ── Divider ──
c.setStrokeColor(HexColor('#262626'))
c.setLineWidth(0.5)
c.line(40, H - 100, W - 40, H - 100)

# ── Order details grid (tx_ref + order_id) ──
c.setFillColor(HexColor('#1a1a1a'))
c.roundRect(40, H - 175, W - 80, 55, 6, fill=1, stroke=0)

c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica-Bold', 8)
c.drawString(55, H - 138, 'TRANSACTION REFERENCE')
c.drawString(W/2 + 15, H - 138, 'ORDER ID')

c.setFillColor(HexColor('#FFFFFF'))
c.setFont('Helvetica-Bold', 11)
c.drawString(55, H - 155, 'TXF-DANNION-PREVIEW')
c.drawString(W/2 + 15, H - 155, 'ORD-2025-XXXXX')

c.setFillColor(HexColor('#6B7280'))
c.setFont('Helvetica', 7)
c.drawString(55, H - 168, 'Flutterwave payment ID')
c.drawString(W/2 + 15, H - 168, 'Internal order reference')

# ── Date + Status row ──
c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica-Bold', 8)
c.drawString(55, H - 200, 'DATE')
c.drawString(W/2 + 15, H - 200, 'STATUS')

c.setFillColor(HexColor('#FFFFFF'))
c.setFont('Helvetica', 10)
c.drawString(55, H - 215, 'July 13, 2026')

c.setFillColor(HexColor('#0AD25A'))
c.setFont('Helvetica-Bold', 10)
c.drawString(W/2 + 15, H - 215, 'FULFILLED')

# ── Divider ──
c.line(40, H - 240, W - 40, H - 240)

# ── Items table ──
c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica-Bold', 8)
c.drawString(55, H - 260, 'ITEM')
c.drawRightString(W - 175, H - 260, 'QTY')
c.drawRightString(W - 40, H - 260, 'AMOUNT')

c.line(40, H - 268, W - 40, H - 268)

c.setFillColor(HexColor('#FFFFFF'))
c.setFont('Helvetica', 11)
c.drawString(55, H - 290, 'UK ISP Proxy')
c.drawRightString(W - 175, H - 290, '1')
c.drawRightString(W - 40, H - 290, 'N 6,500')

# ── Total badge ──
c.setFillColor(HexColor('#0AD25A'))
c.roundRect(W - 175, H - 325, 135, 28, 4, fill=1, stroke=0)
c.setFillColor(HexColor('#000000'))
c.setFont('Helvetica-Bold', 11)
c.drawString(W - 165, H - 315, 'TOTAL')
c.drawRightString(W - 50, H - 315, 'N 6,500')

# ── Credentials card header ──
c.setFillColor(HexColor('#0AD25A'))
c.setFont('Helvetica-Bold', 10)
c.drawString(40, H - 360, 'YOUR PROXY CREDENTIALS')

# ── Credentials card ──
c.setFillColor(HexColor('#1a1a1a'))
c.roundRect(40, H - 480, W - 80, 110, 6, fill=1, stroke=0)

# Username + Password (top row)
c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica-Bold', 8)
c.drawString(55, H - 388, 'USERNAME')
c.drawString(W/2 + 15, H - 388, 'PASSWORD')

c.setFillColor(HexColor('#34D399'))
c.setFont('Helvetica-Bold', 12)
c.drawString(55, H - 405, 'demo_user_4821')
c.drawString(W/2 + 15, H - 405, 'proxyPass_4821!')

# Proxy + Protocol (middle row)
c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica-Bold', 8)
c.drawString(55, H - 425, 'PROXY ADDRESS')
c.drawString(W/2 + 15, H - 425, 'PROTOCOL')

c.setFillColor(HexColor('#34D399'))
c.setFont('Helvetica-Bold', 12)
c.drawString(55, H - 442, '185.199.228.105:8080')
c.drawString(W/2 + 15, H - 442, 'HTTP / SOCKS5')

# Full Format (full width)
c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica-Bold', 8)
c.drawString(55, H - 462, 'FULL FORMAT')

c.setFillColor(HexColor('#D1D5DB'))
c.setFont('Courier', 9)
c.drawString(55, H - 475, 'http://demo_user_4821:proxyPass_4821!@185.199.228.105:8080')

# ── Bottom: Support + Expires ──
c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica-Bold', 8)
c.drawString(40, H - 510, 'EXPIRES')
c.setFillColor(HexColor('#FFFFFF'))
c.setFont('Helvetica', 10)
c.drawString(40, H - 525, 'August 13, 2026')

c.setFillColor(HexColor('#9CA3AF'))
c.setFont('Helvetica-Bold', 8)
c.drawRightString(W - 40, H - 510, 'SUPPORT')
c.setFillColor(HexColor('#FFFFFF'))
c.setFont('Helvetica', 10)
c.drawRightString(W - 40, H - 525, '@StyxproxyBot  |  hello@styxproxy.com')

# ── Footer ──
c.setStrokeColor(HexColor('#262626'))
c.line(40, H - 580, W - 40, H - 580)

c.setFillColor(HexColor('#6B7280'))
c.setFont('Helvetica', 8)
c.drawCentredString(W/2, H - 600, 'Need help? Chat with Charon via @StyxproxyBot')
c.setFont('Helvetica', 7)
c.drawCentredString(W/2, H - 614, 'This receipt was generated automatically. No signature required.')

# ── Bottom accent bar ──
c.setFillColor(HexColor('#0AD25A'))
c.rect(0, 0, W, 4, fill=1, stroke=0)

# Logo watermark (subtle)
c.saveState()
c.translate(W - 110, 50)
c.setFillColor(HexColor('#0AD25A'))
c.setFillAlpha(0.08)
c.drawImage(logo_path, 0, 0, width=80, height=80, mask='auto')
c.restoreState()

c.showPage()
c.save()
print('PDF generated')

File: frontend/gen-pdf.mjs
import { readFileSync } from 'fs';
import { jsPDF } from 'jspdf';

const doc = new jsPDF({ unit: 'mm', format: 'a4' });
const W = doc.internal.pageSize.getWidth();
const H = doc.internal.pageSize.getHeight();

// Read logo as base64
const logoPath = new URL('./public/header-icon.png', import.meta.url);
const logoBase64 = readFileSync(logoPath).toString('base64');

doc.setFillColor(15, 15, 15);
doc.rect(0, 0, W, H, 'F');

doc.setFillColor(16, 185, 129);
doc.rect(0, 0, W, 6, 'F');

// Add actual logo image
doc.addImage(`data:image/png;base64,${logoBase64}`, 'PNG', 20, 18, 12, 12);

// Brand name
doc.setTextColor(245, 245, 245);
doc.setFontSize(18);
doc.setFont('helvetica', 'bold');
doc.text('styxproxy', 35, 25);
doc.setTextColor(115, 115, 115);
doc.setFontSize(9);
doc.setFont('helvetica', 'normal');
doc.text('Anonymous Proxy Service', 35, 30.5);

doc.setTextColor(16, 185, 129);
doc.setFontSize(8);
doc.setFont('helvetica', 'bold');
doc.text('PAYMENT RECEIPT', W - 20, 22, { align: 'right' });
doc.setTextColor(115, 115, 115);
doc.setFontSize(8);
doc.text('styxproxy.com', W - 20, 27, { align: 'right' });

doc.setDrawColor(42, 42, 42);
doc.setLineWidth(0.5);
doc.line(20, 40, W - 20, 40);

let y = 52;
doc.setFillColor(26, 26, 26);
doc.roundedRect(20, y - 6, W - 40, 22, 3, 3, 'F');

doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('TRANSACTION REFERENCE', 25, y);
doc.text('ORDER ID', W / 2 + 5, y);
doc.setTextColor(245, 245, 245);
doc.setFontSize(10);
doc.setFont('helvetica', 'bold');
doc.text('TXF-DANNION-PREVIEW', 25, y + 8);
doc.text('ORD-2025-XXXXX', W / 2 + 5, y + 8);
doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'normal');
doc.text('Flutterwave', 25, y + 14);
doc.text('Internal', W / 2 + 5, y + 14);

y = 82;
doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('DATE', 25, y);
doc.text('STATUS', W / 2 + 5, y);
doc.setTextColor(245, 245, 245);
doc.setFontSize(9);
doc.setFont('helvetica', 'normal');
doc.text(new Date().toLocaleDateString('en-NG', { year: 'numeric', month: 'long', day: 'numeric' }), 25, y + 7);
doc.setTextColor(16, 185, 129);
doc.setFont('helvetica', 'bold');
doc.text('FULFILLED', W / 2 + 5, y + 7);

doc.setDrawColor(42, 42, 42);
doc.line(20, 98, W - 20, 98);

y = 106;
doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('ITEM', 25, y);
doc.text('QTY', W - 48, y, { align: 'right' });
doc.text('AMOUNT', W - 20, y, { align: 'right' });
doc.setDrawColor(42, 42, 42);
doc.line(20, y + 3, W - 20, y + 3);

y += 10;
doc.setTextColor(245, 245, 245);
doc.setFontSize(9);
doc.setFont('helvetica', 'normal');
doc.text('UK ISP Proxy', 25, y);
doc.text('1', W - 48, y, { align: 'right' });
doc.text('N 6,500', W - 20, y, { align: 'right' });

y += 8;
doc.setDrawColor(42, 42, 42);
doc.line(20, y, W - 20, y);
y += 10;

doc.setFillColor(16, 185, 129);
doc.roundedRect(W - 72, y - 6, 52, 12, 2, 2, 'F');
doc.setTextColor(0, 0, 0);
doc.setFontSize(9);
doc.setFont('helvetica', 'bold');
doc.text('TOTAL', W - 67, y - 1);
doc.text('N 6,500', W - 23, y + 3, { align: 'right' });

y += 24;
doc.setTextColor(16, 185, 129);
doc.setFontSize(8);
doc.setFont('helvetica', 'bold');
doc.text('YOUR PROXY CREDENTIALS', 20, y);

doc.setFillColor(26, 26, 26);
doc.roundedRect(20, y + 4, W - 40, 64, 3, 3, 'F');

y += 14;
doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('USERNAME', 28, y);
doc.setTextColor(52, 211, 153);
doc.setFontSize(10);
doc.setFont('helvetica', 'bold');
doc.text('demo_user_4821', 28, y + 7);

doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('PASSWORD', W / 2 + 5, y);
doc.setTextColor(52, 211, 153);
doc.setFontSize(10);
doc.setFont('helvetica', 'bold');
doc.text('proxyPass_4821!', W / 2 + 5, y + 7);

y += 20;
doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('PROXY ADDRESS', 28, y);
doc.setTextColor(52, 211, 153);
doc.setFontSize(10);
doc.setFont('helvetica', 'bold');
doc.text('185.199.228.105:8080', 28, y + 7);

doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('PROTOCOL', W / 2 + 5, y);
doc.setTextColor(52, 211, 153);
doc.setFontSize(10);
doc.setFont('helvetica', 'bold');
doc.text('HTTP / SOCKS5', W / 2 + 5, y + 7);

y += 20;
doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('FULL FORMAT', 28, y);
doc.setTextColor(115, 115, 115);
doc.setFontSize(7.5);
doc.setFont('helvetica', 'normal');
const fmt = 'http://demo_user_4821:proxyPass_4821!@185.199.228.105:8080';
const fmtLines = doc.splitTextToSize(fmt, W - 56);
doc.text(fmtLines, 28, y + 7);

y += 7 + fmtLines.length * 5;
doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'bold');
doc.text('EXPIRES', 28, y);
doc.setTextColor(245, 245, 245);
doc.setFontSize(9);
doc.setFont('helvetica', 'normal');
doc.text('August 13, 2026', 28, y + 6);

const footerY = H - 18;
doc.setDrawColor(42, 42, 42);
doc.line(20, footerY - 6, W - 20, footerY - 6);
doc.setTextColor(115, 115, 115);
doc.setFontSize(7);
doc.setFont('helvetica', 'normal');
doc.text('Need help? Chat with Charon -> @StyxproxyBot  |  hello@styxproxy.com  |  styxproxy.com', W / 2, footerY, { align: 'center' });
doc.setFontSize(6);
doc.text('This receipt was generated automatically. No signature required.', W / 2, footerY + 5, { align: 'center' });

doc.save('./public/sample-receipt.pdf');
console.log('done');
